Core/clim-basic/ports.lisp

```diff
@@ -101,7 +101,15 @@ is T when focused and NIL otherwise. This is a McCLIM extension.")
     (prog1 result
       (unless (eq old-sheet new-sheet)
         (note-input-focus-changed old-sheet nil)
-        (note-input-focus-changed new-sheet t)))))
+        (note-input-focus-changed new-sheet t)
+        (when-let ((top-sheet (get-top-level-sheet new-sheet)))
+          (setf (focused-sheet top-sheet) new-sheet))))))
+
+(defmethod (setf port-keyboard-input-focus) ((sheet top-level-sheet-mixin) port)
+  (if-let ((focus (focused-sheet sheet)))
+    (call-next-method focus port)
```

Comment:
I had to change this to
```lisp
(defmethod (setf port-keyboard-input-focus) ((sheet top-level-sheet-mixin) port)
  (if-let ((focus (focused-sheet sheet)))
    (if (eq focus sheet)
        (call-next-method)
        (setf (port-keyboard-input-focus port) focus))
    (call-next-method)))
```
Otherwise `call-next-method` would be called with a different set of applicable methods.

Reply:
you are right, thank you. fixed